apis/v1alpha1/instrumentation_types.go

```diff
@@ -22,6 +22,12 @@ import (
 
 // InstrumentationSpec defines the desired state of OpenTelemetry SDK and instrumentation.
 type InstrumentationSpec struct {
+	// Selector is the label selector for affected Pods.
+	// Unlike standard label selectors, `nil` means `everything`, and this is also the default.
+	// This may change in a future CRD version.
+	// +optional
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
```

Comment:
Are all pods from all namespaces matched? 

What is the precedence order when annotation `instrumentation.opentelemetry.io/` is also present on the pod/namespace?

Reply:
I believe this feature only comes into play when the annotation is not set and we try to pick an Instrumentation from the namespace the Pod is in. The annotation has precedence. But we should document this behaviour somewhere.

Reply:
> I believe this feature only comes into play when the annotation is not set and we try to pick an Instrumentation from the namespace the Pod is in

https://github.com/open-telemetry/opentelemetry-operator/blob/c65629b3b9fcb722e770eb58b73ea4f5cc8bc63f/pkg/instrumentation/podmutator.go#L368-L377

Did I understand it wrong? I think the premise for this feature to take effect is that the `instrumentation.opentelemetry.io/inject-xx` is equal to true.

Reply:
What I meant to say was that this mechanism only applies if the annotation doesn't specify which instrumentation should be used. Apologies for the confusion.

Reply:
Thanks; I understand that the documentation has been supplemented.